src/languageservice/parser/jsonParser07.ts

```diff
@@ -82,21 +82,30 @@ export interface IProblem {
 }
 
 export abstract class ASTNodeImpl {
-  public abstract readonly type: 'object' | 'property' | 'array' | 'number' | 'boolean' | 'null' | 'string';
+  abstract readonly type: 'object' | 'property' | 'array' | 'number' | 'boolean' | 'null' | 'string';
 
-  public offset: number;
-  public length: number;
-  public readonly parent: ASTNode;
-  public location: string;
+  offset: number;
+  length: number;
+  readonly parent: ASTNode;
+  location: string;
 
   constructor(parent: ASTNode, offset: number, length?: number) {
     this.offset = offset;
     this.length = length;
     this.parent = parent;
+    this.updateParentPosition();
```

Comment:
I am not thrilled that a node needs to update its parent's position. Shouldn't this be the responsibility of the AST converter logic?

Reply:
It should be, but to be compatible with previous AST, we need recalculate parent position, as some child may be have length bigger that new parser provide, and we extend  length in some cases there: https://github.com/redhat-developer/yaml-language-server/pull/507/files#diff-c61722cb0cd0930a1a34772189f9726eca5a9e95b69e23a6b07acfc40d219d39R166